source/extensions/common/workload_discovery/api.cc

```diff
@@ -46,10 +46,22 @@ Istio::Common::WorkloadMetadataObject convert(const istio::workload::Workload& w
   default:
     break;
   }
+  std::string ns = workload.namespace_();
```

Comment:
`const std::string&` to avoid a copy.

Reply:
I think this is necessary if we do need the defaulting logic?

Reply:
I think you can use `string_view` here, and pull out a constant into `constexpr`?

Reply:
I think that worked thanks!